cmd/entrypoint/cmd/root.go

```diff
@@ -0,0 +1,203 @@
+				return err
+			}
+			if err := runCommand(ctx, runnerDir, runSVCCommand); err != nil {
+				return err
+			}
+
+			extend, err := annotatePods(ctx)
+			if err != nil {
+				return err
+			}
+			if err := slackNotify(ctx, extend); err != nil {
+				return err
+			}
+
+			time.Sleep(time.Duration(1<<63 - 1))
+			return nil
+		})
+		well.Stop()
+		return well.Wait()
+	},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+func checkEnvs() error {
+	if len(podName) == 0 {
+		return fmt.Errorf("%s must be set", constants.PodNameEnvName)
+	}
+	if len(podNamespace) == 0 {
+		return fmt.Errorf("%s must be set", constants.PodNamespaceEnvName)
+	}
+	if len(runnerToken) == 0 {
+		return fmt.Errorf("%s must be set", constants.RunnerTokenEnvName)
+	}
+	if len(runnerOrg) == 0 {
+		return fmt.Errorf("%s must be set", constants.RunnerOrgEnvName)
+	}
+	if len(runnerRepo) == 0 {
+		return fmt.Errorf("%s must be set", constants.RunnerRepoEnvName)
+	}
+	if len(extendDuration) == 0 {
+		extendDuration = "20m"
+	}
+
+	// SLACK_AGENT_SERVICE_NAME is optional
+
+	return nil
+}
+
+func runCommand(ctx context.Context, workDir, commandStr string, args ...string) error {
+	command := exec.CommandContext(ctx, commandStr, args...)
+	command.Stdout = os.Stdout
+	command.Stderr = os.Stderr
+	command.Dir = workDir
+	command.Env = removeEnv()
+	if err := command.Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func removeEnv() []string {
+	rmList := []string{
+		constants.PodNameEnvName,
+		constants.PodNamespaceEnvName,
+		constants.RunnerTokenEnvName,
+		constants.RunnerOrgEnvName,
+		constants.RunnerRepoEnvName,
+		constants.SlackAgentServiceNameEnvName,
+	}
+	var removedEnv []string
+	rmMap := make(map[string]struct{})
+	for _, v := range rmList {
+		rmMap[v] = struct{}{}
+	}
+	for _, target := range os.Environ() {
+		keyvalue := strings.SplitN(target, "=", 2)
+		if _, ok := rmMap[keyvalue[0]]; !ok {
+			removedEnv = append(removedEnv, target)
+		}
+	}
+	return removedEnv
+}
+
+func isFileExists(filename string) bool {
+	_, err := os.Stat(filename)
+	return err == nil
+}
+
+func annotatePods(ctx context.Context) (bool, error) {
+	if isFileExists(extendFlagFile) {
+		dur, err := time.ParseDuration(extendDuration)
+		if err != nil {
+			return false, err
+		}
+		fmt.Printf("Annotate pods with the time %s later\n", extendDuration)
+		agent.AnnotateDeletionTime(ctx, podName, podNamespace, time.Now().Add(dur))
+		return true, nil
+	} else {
+		fmt.Println("Annotate pods with current time")
+		agent.AnnotateDeletionTime(ctx, podName, podNamespace, time.Now())
+		return false, nil
+	}
+}
+
+func slackNotify(ctx context.Context, extend bool) error {
+	// TODO: using ctx
+	var jobResult string
+	switch {
+	case isFileExists(failureFlagFile):
+		jobResult = "failure"
```

Comment:
Could you replace it with a constant?
```suggestion
		jobResult = agent.JobResultFailure
```